### PR TITLE
Dev update dynamic properties

### PR DESCRIPTION
**Pull request by Matt-Web**
- Declared properties in classes to prevent dynamic property warnings.
- Allowed default formatting to be applied even when a font is set without valid table definitions.
- Updated tests to fix previously failing tests.

### CODE REVIEW

File: tests/ExtraParagraphTest.php [hunk 1]
```diff
@@ -8,13 +8,14 @@ final class ExtraParagraphTest extends TestCase
...
   {
-    $rtf = file_get_contents("tests/rtf/extra-closing-paragraph.rtf");
+    $rtf = file_get_contents(__DIR__ ."/rtf/extra-closing-paragraph.rtf");
+
     $document = new Document($rtf);
     $formatter = new HtmlFormatter();
     $html = $formatter->Format($document);
 
     $this->assertEquals(
-      '<p><span style="font-weight:bold;font-family:Calibri;font-size:16px;color:#000000;">Conditions<br/></span><span style="font-family:Calibri;font-size:16px;color:#000000;">&#1;Delivery: FCA in our warehouse in Rotterdam<br/>&#1;Lead Time: 25 working days after confirmation, subject to prior sale<br/>&#1;Payment: 60 days after invoice date<br/>&#1;Quote validity: 30 days',
```
Comment (Danielfelgate) [inline tests/ExtraParagraphTest.php:17]: `&#1;` doesn't even seem to be a valid html entity?

File: src/Html/HtmlFormatter.php [hunk 2]
```diff
@@ -472,7 +477,7 @@ protected function FormatText($text)
 
   protected function GetSourceEncoding()
   {
-    if (isset($this->state->font)) {
+    if (!is_null($this->state->font)) {
```
Comment (Danielfelgate) [inline src/Html/HtmlFormatter.php]: Does this need a `?->` optional chain now? isset silences the warning but is_null will raise a warning if `$this->state` is null.

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,13 @@
 # Changelog
 
+## 1.3
+
+### Update 05 June '24:
+- Declared properties in classes to prevent dynamic property warnings.
+- Allowed default formatting to be applied even when a font is set without valid table definitions.
+- Updated tests to fix previously failing tests .
```
Comment (Danielfelgate) [inline CHANGELOG.md]: minor nit space before .

File: tests/ParseSimpleTest.php [hunk 4]
```diff
@@ -8,20 +8,22 @@ final class ParseSimpleTest extends TestCase
...
   {
-    $rtf = file_get_contents("tests/rtf/hello-world.rtf");
+    $rtf = file_get_contents(__DIR__ ."/rtf/hello-world.rtf");
     $document = new Document($rtf);
     $formatter = new HtmlFormatter();
     $html = $formatter->Format($document);    
 
     $this->assertEquals(
-      '<p><span style="font-family:Calibri;font-size:15px;">Hello, world.</span></p>',
+      '<p><span style="font-family:Calibri;font-size:15px;">Hello, world.</span></p><p></p>',
```
Comment (Danielfelgate) [inline tests/ParseSimpleTest.php:26]: Was the extra \<p>\</p> already being generated and causing the test to fail?

**Review by Danielfelgate — APPROVED** [review]
Haven't been able to test this locally alongside Centrik but the changes all make sense. There might be one issue with using is_null but this looks good to go other than that

**Review by Danielfelgate — APPROVED** [review]
Tested and seeing no warning logs now

File: tests/ParseSimpleTest.php (hunk 4 above)
Comment (Matt-Web) [reply tests/ParseSimpleTest.php:26]: Yes already being generated, it's also the case in our existing tests in our app. Perhaps something to look into at some point

File: tests/ExtraParagraphTest.php (hunk 1 above)
Comment (Matt-Web) [reply tests/ExtraParagraphTest.php:17]: Yea it's odd, but pre-existing so ok for now

**Pull request merged**